drivers/ethernet/eth_adin2111.c

```diff
@@ -269,6 +267,9 @@ static inline void adin2111_port_on_phyint(const struct device *dev)
 static void adin2111_offload_thread(const struct device *dev)
 {
 	struct adin2111_data *ctx = dev->data;
+	const struct adin2111_port_config *pcfg = dev->config;
+	const struct adin2111_config *adin_cfg = pcfg->adin->config;
+	bool is_adin2111 = (adin_cfg->id == ADIN2111_MAC);
```

Comment:
There is a bug... 

`adin2111_config` is `dev->config`. The `dev->config` is not `adin2111_port_config`.
Will create separate PR with a fix later today...

Reply:
You're right. First entry pointer passed to the `k_thread_create` is directly the `adin` device. Will fix it asap.

Reply:
Should be fixed with #61596